src/error.ts

```diff
@@ -1,8 +1,8 @@
-import type { FetchRequest, FetchResponse } from "./fetch";
+import type { FetchOptions, FetchRequest, FetchResponse } from "./fetch";
 
 export class FetchError<T = any> extends Error {
   name = "FetchError";
-  request?: FetchRequest;
+  request?: FetchRequest | FetchOptions;
```

Comment:
I think we can have an additional `options: FetchOptions` field

Reply:
> I think we can have an additional `options: FetchOptions` field

Yes, that's more correct.